### PR TITLE
Rename chainID to domainID

### PR DESCRIPTION
**Pull request by waymobetta**
## Description

Replaces all instances of ChainID/chainID with domainID/DomainID, including flags, descriptions, comments, etc.

TODO:
- [x] variables
- [x] functions
- [x] ABI
- [x] Binaries

**NOTE:** Changes made only to core and core-docs. No changes were made to [chainbridge](https://github.com/chainsafe/chainbridge), [cb-sol-cli](https://github.com/ChainSafe/chainbridge-deploy/tree/main/cb-sol-cli) and [chainbridge-docs](https://github.com/chainsafe/chainbridge-docs) for v1; these repos (and their documentation/guides) still use `chainID/chainId`, respectively. New docs for core will be stored in the [develop branch](https://github.com/chainsafe/chainbridge-docs/tree/develop) of chainbridge-docs.

## Related Issue Or Context

Closes: #79 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [ ] I have added tests to cover my changes.
- [ ] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: chains/evm/calls/utils.go [hunk 1]
```diff
@@ -30,13 +30,13 @@ type ChainClient interface {
...
...\"bytes32\"},{\"internalType\":\"bytes32\",\"name\":\"dataHash\",\"type\":\"bytes32\"}],\"name\":\"voteProposal\",\"outputs\":[],\"stateMutability\":\"nonpayable\",\"type\":\"function\"},{\"inputs\":[{\"internalType\":\"uint8\",\"name\":\"domainID\",\"type\":\"uint8\"},{\"internalType\":\"uint64\",\"name\":\"depositNonce\",\"type\":\"uint64\"},{\"internalType\":\"bytes32\",\"name\":\"dataHash\",\"type\":\"bytes32\"}],\"name\":\"cancelProposal\",\"outputs\":[],\"stateMutability\":\"nonpayable\",\"type\":\"function\"},{\"inputs\":[{\"internalType\":\"uint8\",\"name\":\"domainID\",\"type\":\"uint8\"},{\"internalType\":\"uint64\",\"name\":\"depositNonce\",\"type\":\"uint64\"},{\"internalType\":\"bytes\",\"name\":\"data\",\"type\":\"bytes\"},{\"internalType\":\"bytes32\",\"name\":\"resourceID\",\"type\":\"bytes32\"}],\"name\":\"executeProposal\",\"outputs\":[],\"stateMutability\":\"nonpayable\",\"type\":\"function\"},{\"inputs\":[{\"internalType\":\"addresspayable[]\",\"name\":\"addrs\",\"type\":\"address[]\"},{\"internalType\":\"uint256[]\",\"name\":\"amounts\",\"type\":\"uint256[]\"}],\"name\":\"transferFunds\",\"outputs\":[],\"stateMutability\":\"nonpayable\",\"type\":\"function\"}]"
```
Comment (P1sar) [inline chains/evm/calls/utils.go:33]: You renamed it by hands or took ABI from solidity branch with such changes?

Comment (waymobetta) [reply chains/evm/calls/utils.go]: Renamed it by hand. Regenerated ABIs and binaries as ERC20Handler and ERC20MinterPauser contract were also changed to have this chainID/domainID removed from the struct

Comment (waymobetta) [reply chains/evm/calls/utils.go]: Pushed new change

**Comment by P1sar** [comment]
Lets wait with this until we have stable solidity release with corresponding changes

**Pull request merged**